addons/point_of_sale/static/src/app/navbar/navbar.js

```diff
@@ -46,7 +46,7 @@ export class Navbar extends Component {
     }
 
     get customerFacingDisplayButtonIsShown() {
-        return this.pos.config.iface_customer_facing_display;
+        return this.pos.config.iface_customer_facing_display && !this.ui.isSmall;
```

Comment:
it might be better to use from `isMobileOS()` from `web/static/src/core/browser/feature_detection.js`

Reply:
@vlst-odoo 
Changes have been made!